community/cloud-foundation/templates/cloudbuild/tests/integration/cloudbuild.bats

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env bats
+
+source tests/helpers.bash
+
+TEST_NAME=$(basename "${BATS_TEST_FILENAME}" | cut -d '.' -f 1)
+
+# Create a random 10-char string and save it in a file.
+RANDOM_FILE="/tmp/${CLOUD_FOUNDATION_ORGANIZATION_ID}-${TEST_NAME}.txt"
+if [[ ! -e "${RANDOM_FILE}" ]]; then
+    RAND=$(head /dev/urandom | LC_ALL=C tr -dc a-z0-9 | head -c 10)
+    echo ${RAND} > "${RANDOM_FILE}"
+fi
+
+# Set variables based on the random string saved in the file.
+# envsubst requires all variables used in the example/config to be exported.
+if [[ -e "${RANDOM_FILE}" ]]; then
+    export RAND=$(cat "${RANDOM_FILE}")
+    DEPLOYMENT_NAME="${CLOUD_FOUNDATION_PROJECT_ID}-${TEST_NAME}-${RAND}"
+    # Replace underscores in the deployment name with dashes.
+    DEPLOYMENT_NAME=${DEPLOYMENT_NAME//_/-}
+    CONFIG=".${DEPLOYMENT_NAME}.yaml"
+    # Test specific variables
+    export CLOUDBUILD_NAME="test-build-${RAND}"
+    export BUILD_TIMEOUT="500s"
+    export IMAGE_NAME="test-npm-helloworld-${RAND}"
+    export IMAGE_TAG="gcr.io/${CLOUD_FOUNDATION_PROJECT_ID}/${IMAGE_NAME}"
+    export LOGURL_BASE="https://console.cloud.google.com/gcr/builds/"
+fi
+
+########## HELPER FUNCTIONS ##########
+
+function create_config() {
+    echo "Creating ${CONFIG}"
+    envsubst < ${BATS_TEST_DIRNAME}/${TEST_NAME}.yaml > "${CONFIG}"
+}
+
+function delete_config() {
+    echo "Deleting ${CONFIG}"
+    rm -f "${CONFIG}"
+}
+
+function setup() {
+    # Global setup; executed once per test file.
+    if [ ${BATS_TEST_NUMBER} -eq 1 ]; then
+        create_config
+    fi
+
+    # Per-test setup steps.
+}
+
+function teardown() {
+    # Global teardown; executed once per test file.
+    if [[ "$BATS_TEST_NUMBER" -eq "${#BATS_TEST_NAMES[@]}" ]]; then
+        delete_config
+        rm -f "${RANDOM_FILE}"
+        # delete the container image from registry
+        gcloud container images delete ${IMAGE_TAG}:latest -q
+    fi
+
+    # Per-test teardown steps.
+}
+
+########## TESTS ##########
```

Comment:
where are the tests for triggers?

Reply:
Hi Fatima,

There seems be to no way to either `describe` or `list` triggers using `gcloud` - or at least I didn't find any, which is why the test for `triggers` were not included in the `bats` file.

All I could find is, [this](https://cloud.google.com/cloud-build/docs/running-builds/automate-builds) and this part

![image](https://user-images.githubusercontent.com/42005789/47240809-79c17e80-d3b7-11e8-807e-facadffed7b1.png)

But, if you do have any links please do let me know and I can add them.

I'm going through your other comments, should be done hopefully by Monday.
Have a good weekend!

Reply:
looks like we need a FR for this; will add to our TODO list and merge it for now. Thanks Praveen! Good job cleaning up this PR.